### PR TITLE
Expo support

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-838

## What changed (plus any additional context for devs)
This PR adds support for Expo builds. Because of the changes made to the build process, we should consider our other build process legacy at this point. 

### Checklist before reviewing
- [ ] make sure .env is up to date
- [ ] add `RAINBOW_SCRIPTS_TOKEN=github_xxxxxxxx` to .env from 1pass
- [ ] `yarn nuke && yarn fast`
- [ ] `eas login` (reach out if you need access to the expo project)

### To trigger a development build
Depending on what platform you want to build, you can run `yarn eas-build:dev` for both platforms. `yarn eas-build:dev:<PLATFORM>` for a specific platform. (e.g. - `yarn eas-build:dev:ios`)

## Screen recordings / screenshots
Nothing to really screenshot / show here as this is more of a DX / internal improvement

## What to test
There's quite a bit we want to test here before we should consider this mergable. We'll probably want to do a group bug bash session to see if anything funky comes up.

### CODE REVIEW

File: ios/Rainbow.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -733,17 +747,17 @@
...
 			buildConfigurationList = 13B07F931A680F5B00A75B9A /* Build configuration list for PBXNativeTarget "Rainbow" */;
 			buildPhases = (
-				A1E5EC36516AA3B0EE46BE1F /* [CP] Check Pods Manifest.lock */,
+				09027C46E77CF77316A39F45 /* [CP] Check Pods Manifest.lock */,
+				5E71112ED85590D3BF2BCC85 /* [Expo] Configure project */,
 				13B07F871A680F5B00A75B9A /* Sources */,
 				13B07F8C1A680F5B00A75B9A /* Frameworks */,
 				13B07F8E1A680F5B00A75B9A /* Resources */,
 				00DD1BFF1BD5951E006B06BC /* Bundle React Native code and images */,
-				9FF961FEA7AF435FA18ED988 /* Upload Debug Symbols to Sentry */,
```
Comment (dereknelson) [inline ios/Rainbow.xcodeproj/project.pbxproj:741]: removing this temporarily, will follow up with [app 960](https://linear.app/rainbow/issue/APP-960/ensure-sentry-is-configured-and-working)

**Comment by dereknelson** [comment]
as of [this build](https://app.bitrise.io/build/b6cf5b01-a752-4afa-bb6c-b11c50157094), i believe we can remove the DO NOT MERGE tag

**Comment by dereknelson** [comment]
we've re-added sentry support, and both dev & prod builds have completed successfully!

this PR is reviewable (just download the testflight at 1.9.14 (5) when it's done processing and tap around to ensure nothing obscure broke) and more importantly mergeable!

**Review by dereknelson — APPROVED** [review]
v1.9.14 (8) on testflight is the latest expo build, we're good to go 👍🏻

**Comment by dereknelson** [comment]
undid sentry expo. no errors happening & successful source maps upload

**Review by ibrahimtaveras00 — APPROVED** [review]
Ran https://www.notion.so/rainbowdotme/Critical-Path-Checklist-a907ffa19e854e5492fcefd2bf79f8cd regression on both OS's and everything looks gucci ✅

**Review by skylarbarrera — APPROVED** [review]
merge

**Comment by walmat** [comment]
@SocketSecurity ignore npm/esbuild@0.11.23

**Pull request closed without merging**